


! A. SOURCE

- Article: Empower Your Workflow: Harnessing the Power of LM Studio and Ollama for Seamless Local LLM Execution 
https://flaven.fr/2023/12/empower-your-workflow-harnessing-the-power-of-lm-studio-and-ollama-for-seamless-local-llm-execution/

- Path (local)
cd /Users/brunoflaven/Documents/03_git/ia_usages/ia_using_other_llm/

! B. DIRECTORIES ON GITHUB
Source: https://github.com/bflaven/ia_usages/tree/main/ia_using_other_llm

! C. TOOLS & ENVIRONMENT

- Tools to create and manage of virtual environments:

--- ANACONDA
https://www.anaconda.com/

--- PYTHON-POETRY
https://python-poetry.org/

--- VENV
https://docs.python.org/3/library/venv.html

- The framework to create an API in Python

--- ollama.ai
https://ollama.ai/

--- lmstudio.ai
https://lmstudio.ai/

--- for LLMs huggingface.co
https://huggingface.co/

--- MANAGE ENVIRONMENT ANACONDA
[env]
# Conda Environment
conda create --name langchain_ai python=3.9.13
conda info --envs
source activate langchain_ai
conda deactivate

# if needed to remove
conda env remove -n [NAME_OF_THE_CONDA_ENVIRONMENT]
conda env remove -n langchain_ai

# update conda 
conda update -n base -c defaults conda

# to export requirements
pip freeze > requirements.txt

# to install
pip install -r requirements.txt

conda install -c conda-forge langchain


# [path]
cd /Users/brunoflaven/Documents/03_git/ia_usages/ia_using_other_llm/


# launch the file
python samwit_basic_chain.py


# EXPLICATIONS
https://github.com/samwit/langchain-tutorials/blob/382e8db4dc5e01fc400bee8d4146cb1a2e9c3150/ollama/basic.py

# COMMANDS FOR OLLAMA

# To run and chat with Llama 2
ollama run llama2
ollama run llama2-uncensored
ollama run orca-mini


# remove a model
ollama rm llama2
ollama rm orca-mini
ollama rm mistral
ollama rm falcon:7b
ollama rm mistral:text
ollama rm llama2:latest
ollama rm orca-mini:latest

# list the model
ollama list

# when you are in the model you can use
>>> /?
>>> /list
>>> /set verbose

# to get out from the model
/exit


! D. EXAMPLES


+ OBJECTIVE
I want to find a way to locally test other LLMs in a more confidential and secure way than on "classical" websites such as ChatGPT (https://chat.openai.com/) or the Microsoft’s copilot (https://copilot.microsoft.com). Especially, if I start to work with more personal data.


+ 1. Using Ollama:
--- ia_using_other_llm_ollama_1.mov

- DESCRIPTION
I have download ollama at https://ollama.ai/download
Check the models of LLMs avalaible at https://ollama.ai/library
And... I try to use this new LLM.

- CONSOLE_SCREEN (black)

# GO TO DIR
cd /Users/brunoflaven/Documents/03_git/ia_usages/ia_using_other_llm

# List LLMS
ollama list

# Run for orca-mini
ollama run orca-mini

# launch orca-mini and langchain
python samwit_basic.py
python samwit_basic_chain.py

+ 2. Using Ollama:
--- ia_using_other_llm_lmstudio_2.mov

- DESCRIPTION
Need to install lmstudio.ai from https://lmstudio.ai/ (Mac, PC, Linux). Non console required. You can look for any LLMs.

- NO CONSOLE_SCREEN

- TIP 
Take some examples from https://github.com/bflaven/ia_usages/tree/main/ai_chatgpt_usages


- DEMO
--- 1. Let search for llama2, orca, falcon, mistral... etc

--- 2. Look for mistral and install it TheBloke/Mistral-7B-Instruct-v0.1-GGUF/mistral-7b-instruct-v0.1.Q2_K.gguf. Check https://huggingface.co/TheBloke/Mistral-7B-Instruct-v0.1-GGUF


--- 3. Use Mistral-7B-Instruct-v0.1 for different purposes

See more examples in
https://github.com/bflaven/ia_usages/tree/main/ai_chatgpt_usages


! E. VIDEOS
- 001_ia_using_other_llm_ollama.mov
001_ia_using_other_llm_ollama.png

Main title: Using LLM with Ollama or LM Studio



--- BF (human)
Make a more sexy description and provide a title from this text: 
Download the soft ollama at https://ollama.ai/download
Check the models of LLMs avalaible at https://ollama.ai/library and... I try to use this new LLM in the Mac terminal


- Discover LLMs potential (orca-mini) help through Ollama for daily tasks in a safe and confidential way.


Discover LLMs potential (orca-mini) help 
through Ollama for daily tasks 
in a safe and confidential way.

--- IA (chatgpt)
Title: "Unleash Sensual Linguistic Mastery: Dive into the Allure of Ollama's Sultry Software"

Description:
Embark on a journey of linguistic seduction with the irresistibly alluring Ollama AI. Download the soft Ollama at https://ollama.ai/download and immerse yourself in a world where language becomes a sensual dance of intellect and sophistication. Explore the enchanting array of Language Models (LLMs) available at https://ollama.ai/library, each a tantalizing companion in the realm of linguistic passion. Feel the pulse of desire as you wield this new LLM, effortlessly commanding its powers in the Mac terminal. Let Ollama awaken the seductive poet within you, turning every command into a symphony of sophistication and style. Unleash the power of words with Ollama—where intelligence meets allure.

--- IA (mistral)



- 002_ia_using_other_llm_ollama.mov
--- 002a_ia_using_other_llm_ollama.mov
--- 002b_ia_using_other_llm_ollama.mov
--- 002c_ia_using_other_llm_ollama.mov
002_ia_using_other_llm_ollama.png


--- BF (human)
Make a more sexy description and provide a title from this text:
Using a LLM like Mistral-7B-Instruct-v0.1 to convert some coding from one format to another example UAT test written in Cypress convert into gherkin feature.

- Discover LLMs (Mistral) potential 
help through LM Studio for daily tasks 
in a safe and confidential way.


--- IA (chatgpt)
Title: "Coding Elegance Redefined: Unleash Passion in Syntax with Mistral-7B-Instruct-v0.1"

Description:
Indulge in the art of coding seduction as Mistral-7B-Instruct-v0.1 transforms the mundane into a tantalizing dance of syntax. Picture this: the sultry conversion of a UAT test, written in Cypress, into the rhythmic allure of a Gherkin feature. Let Mistral-7B-Instruct-v0.1 be your guide in this intimate affair of code translation. Sensually traverse the landscape of formats, as lines of code intertwine like the steps of a passionate tango, creating a masterpiece that transcends the ordinary. Explore the boundaries of coding elegance, where Mistral-7B-Instruct-v0.1 reigns supreme, turning the technical into the downright seductive. Elevate your coding experience, let desire dictate the syntax, and succumb to the irresistible charm of Mistral-7B-Instruct-v0.1.

